builtin/providers/aws/resource_aws_api_gateway_method.go

```diff
@@ -0,0 +1,221 @@
+				Required: true,
+				ForceNew: true,
+			},
+
+			"authorization": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"api_key_required": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+
+			"request_models": &schema.Schema{
+				Type:     schema.TypeMap,
+				Optional: true,
+				Elem:     schema.TypeString,
+			},
+		},
+	}
+}
+
+func resourceAwsApiGatewayMethodCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).apigateway
+
+	models := make(map[string]string)
+	for k, v := range d.Get("request_models").(map[string]interface{}) {
+		models[k] = v.(string)
+	}
+
+	parameters := make(map[string]bool)
+	if parameterData, ok := d.GetOk("request_parameters"); ok {
+		params := parameterData.(*schema.Set).List()
+		for k := range params {
+			parameters[params[k].(string)] = true
+		}
+	}
+
+	_, err := conn.PutMethod(&apigateway.PutMethodInput{
+		AuthorizationType: aws.String(d.Get("authorization").(string)),
+		HttpMethod:        aws.String(d.Get("http_method").(string)),
+		ResourceId:        aws.String(d.Get("resource_id").(string)),
+		RestApiId:         aws.String(d.Get("api_id").(string)),
+		RequestModels:     aws.StringMap(models),
+		RequestParameters: nil,
+		ApiKeyRequired:    aws.Bool(d.Get("api_key_required").(bool)),
+	})
+	if err != nil {
+		return fmt.Errorf("Error creating API Gateway Method: %s", err)
+	}
+
+	d.SetId(fmt.Sprintf("%s-%s-%s", d.Get("api_id").(string), d.Get("resource_id").(string), d.Get("http_method").(string)))
+	log.Printf("[DEBUG] API Gateway Method ID: %s", d.Id())
+
+	return nil
+}
+
+func resourceAwsApiGatewayMethodRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).apigateway
+
+	log.Printf("[DEBUG] Reading API Gateway Method %s", d.Id())
+	out, err := conn.GetMethod(&apigateway.GetMethodInput{
+		HttpMethod: aws.String(d.Get("http_method").(string)),
+		ResourceId: aws.String(d.Get("resource_id").(string)),
+		RestApiId:  aws.String(d.Get("api_id").(string)),
+	})
+	if err != nil {
+		d.SetId("")
+		return err
+	}
+	log.Printf("[DEBUG] Received API Gateway Method: %s", out)
+	d.SetId(fmt.Sprintf("%s-%s-%s", d.Get("api_id").(string), d.Get("resource_id").(string), d.Get("http_method").(string)))
+
+	return nil
+}
+
+func resourceAwsApiGatewayMethodUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).apigateway
+
+	log.Printf("[DEBUG] Reading API Gateway Method %s", d.Id())
+	operations := make([]*apigateway.PatchOperation, 0)
+	if o, n := d.GetChange("resource_id"); o.(string) != n.(string) {
+		operations = append(operations, &apigateway.PatchOperation{
+			Op:    aws.String("replace"),
+			Path:  aws.String("/resourceId"),
+			Value: aws.String(d.Get("resource_id").(string)),
+		})
+	}
+
+	if d.HasChange("request_models") {
+		o, n := d.GetChange("request_models")
+		oP := o.(map[string]interface{})
```

Comment:
Similarly as above, would you mind decoupling the logic of add/removal detection into a function and putting it into `builtin/providers/aws/structure.go`?